src/Microsoft.TestPlatform.Common/Utilities/AssemblyResolver.cs

```diff
@@ -53,13 +57,11 @@ public AssemblyResolver(IEnumerable<string> directories)
             {
                 this.searchDirectories = new HashSet<string>(directories);
             }
-#if NET46
-            AppDomain.CurrentDomain.AssemblyResolve += new ResolveEventHandler(OnResolve);
-#else
-            AssemblyLoadContext.Default.Resolving += this.OnResolve;
-#endif
-        }
 
+            this.platformAssemblyResolver = new PlatformAssemblyResolver();
```

Comment:
Given there is a PlatformAssemblyResolver, I think we should add unit tests for the business logic in `TestPluginCache` and `AssemblyResolver`.